### PR TITLE
Feat navsys graph storage

### PR DESCRIPTION
**Pull request by JeremyRuizenaar**
Bij deze een pull request voor the graph factory en user input.

Klassen toegevoegd voor deze feature
-graphinput, graph factory. comments

Leave some feedback

### CODE REVIEW

File: modules/NAVSYS/src/graph.cc [hunk 1]
```diff
@@ -10,28 +10,95 @@ void Graph::addNode(const Node node) {
...
+    it = std::find_if(std::begin(nodes), std::end(nodes),[&](const Node & node) -> bool{
+        return node.getNodeName() == name;
+    });
+    return it;
+}
+
+void Graph::dumpGraphToDisk(const std::string nodeFilePath,const std::string verticeFilePath) {
+
+    FILE* nodeFile;
+    nodeFile = fopen(nodeFilePath.c_str(),"w");
```
Comment (LuukSteeman) [inline modules/NAVSYS/src/graph.cc]: Got this warning

Severity	Code	Description	Project	File	Line	Suppression State
Warning	C4996	'fopen': This function or variable may be unsafe. Consider using fopen_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details.	navsys	C:\Users\luuks\OneDrive\Documenten\HBO-ICT_jaar_2\Periode 4\R2D2\R2D2-2017\modules\NAVSYS\src\graph.cc	64

File: modules/NAVSYS/src/graph.cc [hunk 2]
```diff
@@ -10,28 +10,95 @@ void Graph::addNode(const Node node) {
...
+        tmp+="]";
+        tmp+="\n";
+        std::cout << tmp;
+        fputs(tmp.c_str(),nodeFile );
+    });
+
+    fclose(nodeFile);
+
+    FILE* verticeFile;
+    verticeFile = fopen(verticeFilePath.c_str(),"w");
```
Comment (LuukSteeman) [inline modules/NAVSYS/src/graph.cc]: Same warning

Severity	Code	Description	Project	File	Line	Suppression State
Warning	C4996	'fopen': This function or variable may be unsafe. Consider using fopen_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details.	navsys	C:\Users\luuks\OneDrive\Documenten\HBO-ICT_jaar_2\Periode 4\R2D2\R2D2-2017\modules\NAVSYS\src\graph.cc	64

**Review by LuukSteeman — COMMENTED** [review]
We use a different commenting norm according to chris.
Please replace all comments starting with `///` to ```/*
*
*/```

**Comment by LuukSteeman** [comment]
had to add
`#include <string>`
to prevent bunch of errors in node.hh graphfactory.hh and graph.hh

File: modules/NAVSYS/CMakeLists.txt [hunk 3]
```diff
@@ -10,6 +10,9 @@ include_directories(${CMAKE_CURRENT_SOURCE_DIR}/src)
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/include)
 
 set(sources
-    src/main.cc src/node.cc src/node.hh src/coordinate.cc src/coordinate.hh src/vertice.cc src/vertice.hh src/graph.cc src/graph.hh)
+
```
Comment (sqrtroot) [inline modules/NAVSYS/CMakeLists.txt]: Too many whitespaces

File: modules/NAVSYS/src/coordinate.hh [hunk 4]
```diff
@@ -6,6 +6,7 @@
 #define NAVIGATIONSYSTEM_COORDINATE_HH
```
Comment (sqrtroot) [inline modules/NAVSYS/src/coordinate.hh:6]: A lot of indentation changes in not changed files

File: modules/NAVSYS/src/graph-factory.cc [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+
+std::vector<Node>::iterator GraphFactory::getNodeByName(std::string name, std::vector<Node> nodes) {
+    std::vector<Node>::iterator it;
+    it = std::find_if(std::begin(nodes), std::end(nodes),[&](const Node & node) -> bool{
+        return node.getNodeName() == name;
+    });
+    return it;
+}
+
+/*author jeremy
```
Comment (sqrtroot) [inline modules/NAVSYS/src/graph-factory.cc]: Documentation in .hh

File: modules/NAVSYS/src/graph-factory.hh [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+#ifndef NAVSYS_GRAPHFACTORY_HH
+#define NAVSYS_GRAPHFACTORY_HH
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <vector>
+#include "graph.hh"
+#include "node.hh"
+
+using namespace std;
```
Comment (sqrtroot) [inline modules/NAVSYS/src/graph-factory.hh]: No using namespace std

File: modules/NAVSYS/src/graph-factory.hh [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+class GraphFactory {
+private:
+
+
+/**
+ * \brief    This function finds a node in the node  vector by name
+ * \param[in] name name of node
+ * \return  it  iterator to node element
+ */
+    std::vector<Node>::iterator getNodeByName(std::string name, std::vector<Node> nodes);
```
Comment (sqrtroot) [inline modules/NAVSYS/src/graph-factory.hh]: Why return the iterator and not the node itself

File: modules/NAVSYS/src/graph-factory.cc [hunk 8]
```diff
@@ -0,0 +1,213 @@
...
+ */
+#include "graph-factory.hh"
+#include <algorithm>
+
+
+
+
+Graph* GraphFactory::createGraph(std::string nodeFilePath,std::string verticeFilePath ){
+
+    Graph * graph = new Graph();
```
Comment (sqrtroot) [inline modules/NAVSYS/src/graph-factory.cc]: Your documentation(the one that is missing) doesn't show the use of new in this function, this probably is going to create memory leaks. Either document this very clear or change the use of new

File: modules/NAVSYS/src/graph-input.cc [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+ * \file
+ * \brief     This file contains the implementations of the graphIO class,
+ *            which is the class that handles all user input in regards to a graph.
+ * \author    jeremy
+ * \copyright Copyright (c) 2017, The R2D2 Team
+ * \license   See LICENSE
+ */
+#include "graph-input.hh"
+
+void GraphInput::getNodeEntryFromScreen(Graph * graph) {
```
Comment (sqrtroot) [inline modules/NAVSYS/src/graph-input.cc]: Why not use references, raw pointers are unsafe

File: modules/NAVSYS/src/graph.cc [hunk 10]
```diff
@@ -10,28 +10,95 @@ void Graph::addNode(const Node node) {
...
+
+std::vector<Node>::iterator Graph::getNodeByName(std::string name) {
+    std::vector<Node>::iterator it;
+    it = std::find_if(std::begin(nodes), std::end(nodes),[&](const Node & node) -> bool{
+        return node.getNodeName() == name;
+    });
+    return it;
+}
+
+void Graph::dumpGraphToDisk(const std::string nodeFilePath,const std::string verticeFilePath) {
```
Comment (sqrtroot) [inline modules/NAVSYS/src/graph.cc]: Is this a task of the graph or of the factory or of something else? plz discus

File: modules/NAVSYS/src/graph.hh [hunk 11]
```diff
@@ -5,34 +5,63 @@
...
  * \brief Graph class
  * \details
  * Container class for nodes and vertices in the same graph
  */
+
 class Graph {
 private:
-  std::vector<Node> nodes;
-  std::vector<Vertice> vertices;
+    std::vector<Node> nodes;
```
Comment (sqrtroot) [inline modules/NAVSYS/src/graph.hh:22]: Again a lot of white space changes in unrelated files

**Review by sqrtroot — CHANGES_REQUESTED** [review]
Some small changes and next time please have a look at what you actually commit. If you change white space because it's wrong please make it a commit so you've stated your intentions

**Comment by sqrtroot** [comment]
To change the unsave fopen warning you could use ifstreams

File: modules/NAVSYS/src/graph-input.cc (hunk 9 above)
Comment (JeremyRuizenaar) [reply modules/NAVSYS/src/graph-input.cc]: fixed

File: modules/NAVSYS/src/graph-factory.hh (hunk 6 above)
Comment (JeremyRuizenaar) [reply modules/NAVSYS/src/graph-factory.hh]: fixed

File: modules/NAVSYS/src/graph-factory.cc (hunk 5 above)
Comment (JeremyRuizenaar) [reply modules/NAVSYS/src/graph-factory.cc]: fixed

File: modules/NAVSYS/src/graph.hh (hunk 11 above)
Comment (JeremyRuizenaar) [reply modules/NAVSYS/src/graph.hh:22]: noticed.

File: modules/NAVSYS/src/graph-factory.hh (hunk 7 above)
Comment (JeremyRuizenaar) [reply modules/NAVSYS/src/graph-factory.hh]: I did not know this was possible, fixed

File: modules/NAVSYS/src/graph.cc (hunk 10 above)
Comment (JeremyRuizenaar) [reply modules/NAVSYS/src/graph.cc]: The factory only creates the graph. I have put the responsibility of storing the graph at the graph itself.

 Back when  i worked on it i  did not see any reason to have the factory do anything else besides creating the graph.

File: modules/NAVSYS/src/graph.cc (hunk 2 above)
Comment (JeremyRuizenaar) [reply modules/NAVSYS/src/graph.cc]: fixed

File: modules/NAVSYS/CMakeLists.txt (hunk 3 above)
Comment (JeremyRuizenaar) [reply modules/NAVSYS/CMakeLists.txt]: fixed

File: modules/NAVSYS/src/graph-factory.cc (hunk 8 above)
Comment (JeremyRuizenaar) [reply modules/NAVSYS/src/graph-factory.cc]: Removed the use of new, but stumbled across a different problem of the created graph being destroyed at the end of the function. I have circumvented this by doing the following:
GraphFactory factory = GraphFactory();               
Graph g = Graph();                                   
factory.createGraph(nodeFilePath,verticeFilePath, g); 

This however required the graph to be initialized(empty) and then passed to the createGraph() function.
Let me know if this is allright.

**Review by sqrtroot — APPROVED** [review]

**Pull request merged**